test/test_torch.py

```diff
@@ -8454,29 +8454,40 @@ def test_scatter_add_bool(self):
 
     def test_masked_scatter(self):
         with warnings.catch_warnings(record=True) as w:
-            for dtype in [torch.uint8, torch.bool]:
-                num_copy, num_dest = 3, 10
-                dest = torch.randn(num_dest)
-                src = torch.randn(num_copy)
-                mask = torch.tensor((0, 0, 0, 0, 1, 0, 1, 0, 1, 0), dtype=dtype)
-                dest2 = dest.clone()
-                dest.masked_scatter_(mask, src)
-                j = 0
-                for i in range(num_dest):
-                    if mask[i]:
-                        dest2[i] = src[j]
-                        j += 1
-                self.assertEqual(dest, dest2, 0)
-
-                # make source bigger than number of 1s in mask
-                src = torch.randn(num_dest)
-                dest.masked_scatter_(mask, src)
-
-                # make src smaller. this should fail
-                src = torch.randn(num_copy - 1)
-                with self.assertRaises(RuntimeError):
+            for maskType in [torch.uint8, torch.bool]:
+                for dt in torch.testing.get_all_dtypes():
+                    num_copy, num_dest = 3, 10
+                    dest = torch.tensor([1, 2, 3, 4, 5, 6, 7, 8, 9, 10], dtype=dt)
+                    dest2 = dest.clone()
+                    src = torch.tensor([0, 0, 0, 0, 0, 0, 0, 0, 0, 0], dtype=dt)
+                    mask = torch.tensor((0, 0, 0, 0, 1, 0, 1, 0, 1, 0), dtype=maskType)
+
+                    if dt == torch.bool:
+                        # torch.bool is a special case and is being tested
+                        # in a separate test
```

Comment:
...because bool will truncate elements outside of its range? (Spell it out for the reader ;)

Reply:
...but that's not actually all that special :).  I don't know why we just don't have an if statement that sets up the destination for the case where it's bool -- we don't test the edge cases for bool tensors for what seems like non-compelling reasons.

Reply:
I find this a bit more cleaner. in case when there are a lot of if/else conditions, its hard to figure out what does the test do quickly. But if you insist, i will combine all masked_* tests with their masked_*_bool versions